plugins/notifications-backend/src/service/handlers.ts

```diff
@@ -0,0 +1,407 @@
+/* eslint-disable func-names */
+import { CatalogClient } from '@backstage/catalog-client';
+
+import { Knex } from 'knex';
+
+import { ActionsInsert, MessagesInsert } from './db';
+import {
+  CreateNotificationRequest,
+  DefaultMessageScope,
+  DefaultOrderBy,
+  DefaultOrderDirection,
+  DefaultPageNumber,
+  DefaultPageSize,
+  DefaultUser,
+  Notification,
+  NotificationsFilterRequest,
+  NotificationsOrderByDirections,
+  NotificationsOrderByFields,
+  NotificationsSortingRequest,
+} from './types';
+
+// createNotification
+// returns string id of created notification
+export async function createNotification(
+  dbClient: Knex<any, any>,
+  catalogClient: CatalogClient,
+  req: CreateNotificationRequest,
+): Promise<{ messageId: string }> {
+  let isUser = false;
+
+  // validate users
```

Comment:
I would like to see that the users of target groups are just appended to the targetUsers and the groups are not even saved to the database. This allows to only target notifications to specific users and that way it's possible to mark the status for the notifications as in `new`, `read`, `discarded` etc.